### PR TITLE
[BUGFIX] `argilla`: support reading datasets without distribution

### PR DESCRIPTION
**Pull request by frascuchon**
# Description

This PR fixes errors when loading dataset settings and the optional distribution model is `None` (thanks @jfcalvo for reporting this).

Also, prevent mutating the default task distribution by defining a default factory instead of a default value.

**Type of change**

- Bug fix (non-breaking change which fixes an issue)

**How Has This Been Tested**

**Checklist**

- I added relevant documentation
- I followed the style guidelines of this project
- I did a self-review of my code
- I made corresponding changes to the documentation
- I confirm My changes generate no new warnings
- I have added tests that prove my fix is effective or that my feature works
- I have added relevant notes to the CHANGELOG.md file (See https://keepachangelog.com/)

### CODE REVIEW

File: argilla/tests/unit/test_settings/test_settings.py [hunk 1]
```diff
@@ -154,6 +154,15 @@ def test_settings_access_by_out_of_range(self):
         with pytest.raises(IndexError):
             _ = settings.fields[10]
 
+    def test_settings_with_modified_default_task_distribution(self):
```
Comment (davidberenstein1957) [inline argilla/tests/unit/test_settings/test_settings.py:160]: @frascuchon this test case seems unrelated to this PR?

We want to test if the code is backward compatible with existing datasets without distributions and ensure we don't overwrite them or get an error, correct?

Comment (frascuchon) [reply argilla/tests/unit/test_settings/test_settings.py:160]: Yes. I missed uploading the test. Thanks!

File: argilla/src/argilla/settings/_resource.py [hunk 2]
```diff
@@ -133,7 +133,7 @@ def allow_extra_metadata(self, value: bool):
 
     @property
     def distribution(self) -> TaskDistribution:
-        return self._distribution
+        return self._distribution or TaskDistribution.default()
```
Comment (frascuchon) [inline argilla/src/argilla/settings/_resource.py:136]: This change prevents to mutate the default task distribution value

**Review by jfcalvo — APPROVED** [review]
